helpers/aave/index.ts

```diff
@@ -0,0 +1,140 @@
+  const lendingPoolContract: Interface = new Interface([
+    AaveAbis.reserveDataUpdatedEvent,
+  ])
+  const events: Array<ReserveDataUpdatedEvent> = (await options.getLogs({
+    target: pool.lendingPoolProxy,
+    flatten: false,
+    entireLog: true,
+    topic: '0x804c9b842b2748a22bb64b345453a3de7ca54a6ca45ce00d415894979e22897a',
+  }))
+  .map((log: any) => {
+    const decodeLog: any = lendingPoolContract.parseLog(log);
+
+    const event: ReserveDataUpdatedEvent = {
+      transactionHash: log.transactionHash,
+      blockNumber: Number(log.blockNumber),
+      reserve: normalizeAddress(decodeLog.args[0]),
+      liquidityIndex: BigInt(decodeLog.args[4].toString()),
+    }
+
+    return event
+  })
+  .flat()
+
+  for (const event of events) {
+    const reserveData = await options.api.call({
+      abi: pool.version === 3 ? AaveAbis.getReserveDataV3 : AaveAbis.getReserveDataV2,
+      target: pool.dataProvider,
+      params: [event.reserve],
+      skipCache: true,
+      permitFailure: false,
+      block: event.blockNumber - 1,
+    })
+
+    let liquidityIndex = BigInt(0)
+    let totalLiquidity = BigInt(0)
+    if (pool.version === 3) {
+      liquidityIndex = BigInt(reserveData[9].toString())
+      totalLiquidity = BigInt(reserveData[2].toString())
+    } else {
+      liquidityIndex = BigInt(reserveData[7].toString())
+      totalLiquidity = BigInt(reserveData[0].toString()) + BigInt(reserveData[1].toString()) + BigInt(reserveData[2].toString())
+    }
+
+    const growthLiquidityIndex = event.liquidityIndex - liquidityIndex
+    const interestAccrued = totalLiquidity * growthLiquidityIndex / LiquidityIndexDecimals
+    const revenueAccrued = reserveFactors[event.reserve] * interestAccrued / ReserveFactorDecimals
+
+    dailyFees.add(event.reserve, interestAccrued.toString())
+    dailyRevenue.add(event.reserve, revenueAccrued.toString())
+  }
+
+  return {
+    dailyFees,
+    dailyRevenue,
+  }
+}
+
+export function aaveExport(config: IJSON<Array<AaveLendingPoolConfig>>) {
+  const exportObject: BaseAdapter = {}
+  Object.entries(config).map(([chain, pools]) => {
+    exportObject[chain] = {
+      fetch: (async (options: FetchOptions) => {
+        const totalDailyFees = options.createBalances()
+        const totalDailyRevenue = options.createBalances()
+        const totalDailyHoldersRevenue = options.createBalances()
+        const totalDailySupplySideRevenue = options.createBalances()
+
+        for (const pool of pools) {
+          const { dailyFees, dailyRevenue } = await getPoolFees(pool, options)
+
+          totalDailyFees.addBalances(dailyFees)
+          totalDailyRevenue.addBalances(dailyRevenue)
+          totalDailySupplySideRevenue.addBalances(dailyFees)
+          totalDailyHoldersRevenue.addBalances(dailyRevenue)
+
+          Object.entries(dailyRevenue.getBalances()).forEach(([token, balance]) => {
+            totalDailySupplySideRevenue.addTokenVannila(token, Number(balance) * -1)
```

Comment:
you can simplify this by cloning the dailyFees object (`.clone()` ) and then subtracting daily revenue object (https://github.com/DefiLlama/defillama-sdk/blob/master/src/Balances.ts#L153)